Convert Policy, ShouldAllowFromThirdParty enums in SecurityOrigin.h to an enum class

AlexsanderDamaceno opened the pull request:
#### fe533bd6abf71accf875509f9d9e92e44d5bd312
<pre>
Convert Policy, ShouldAllowFromThirdParty enums in SecurityOrigin.h to an enum class
<a href="https://bugs.webkit.org/show_bug.cgi?id=281104">https://bugs.webkit.org/show_bug.cgi?id=281104</a>

Reviewed by Sihui Liu.

Convert Policy, ShouldAllowFromThirdParty enums in SecurityOrigin.h to an enum class

* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::SecurityOrigin::canShowNotifications const):
* Source/WebCore/page/SecurityOrigin.h:

Canonical link: <a href="https://commits.webkit.org/284870@main">https://commits.webkit.org/284870@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/04d0bc7f823d1380139274fdfe5f5ede3a22e600

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/70790 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/50200 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/23559 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/74887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/21990 "Failed to checkout and rebase branch from PR 34880") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/72906 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/57998 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/21811 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/74887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/59/builds/21990 "Failed to checkout and rebase branch from PR 34880") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/73856 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/45633 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61030 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/74887 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/42288 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/18463 "Passed tests") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/20332 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/64223 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/18825 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/76601 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/15020 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/18026 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/63765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/15064 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/61093 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63711 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15669 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11783 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/5431 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/46001 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/772 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/47073 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/48354 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/46815 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

szewai commented on Source/WebCore/page/SecurityOrigin.h:
```diff
@@ -40,7 +40,7 @@ class OriginAccessPatterns;
 class SecurityOrigin : public ThreadSafeRefCounted<SecurityOrigin> {
 public:
-    enum Policy {
+    enum class Policy {
```
We prefer specifying underlying type:
```
enum class Policy : uint8_t {
...
}
```

AlexsanderDamaceno replied:
@szewai Changed.

AlexsanderDamaceno committed:
Convert Policy, ShouldAllowFromThirdParty enums in SecurityOrigin.h to an enum class
https://bugs.webkit.org/show_bug.cgi?id=281104

Reviewed by Sihui Liu.

Convert Policy, ShouldAllowFromThirdParty enums in SecurityOrigin.h to an enum class

* Source/WebCore/page/SecurityOrigin.cpp:
(WebCore::SecurityOrigin::canShowNotifications const):
* Source/WebCore/page/SecurityOrigin.h:

Canonical link: https://commits.webkit.org/284870@main
